src/rest_vol_file.c

```diff
@@ -653,10 +656,7 @@ RV_file_close(void *file, hid_t dxpl_id, void **req)
     if (H5I_FILE != _file->obj_type)
         FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "not a file");
 
-    if (_file->u.file.filepath_name) {
-        RV_free(_file->u.file.filepath_name);
-        _file->u.file.filepath_name = NULL;
-    }
+    _file->u.file.ref_count--;
 
     if (_file->u.file.fapl_id >= 0) {
```

Comment:
If we're doing proper ref. counting, then we probably want to move closing of any file fields like these into the `_file->u.file.ref_count == 0` block as well.